kb/how-to-troubleshoot-last-connection-error.md

```diff
@@ -0,0 +1,78 @@
+---
+slug: troubleshoot-last-connection-error
+title: Troubleshoot the "failed reaching server" error
+tags:
+  - cloud
+  - namespaces
+date: 2023-05-19T00:00:00Z
+---
+
+The message `Failed reaching server: last connection error` can often result from an expired TLS certificate or during the Server startup process, in which the Client requests reach the Server before the roles are fully initialized.
+
+This troubleshooting guide shows you how to do the following:
+
+- Verify the certification expiration date
+- Renew the certification
+- Update the server configuration
+
+<!-- truncate -->
+
+### Verify TLS certification expiration date
+
+The first step in troubleshooting this error is to verify the expiration date of the TLS certification.
+
+**Existing certificate management infrastructure**
+
+If you are using an existing certificate management infrastructure, use it to verify the TLS expiration date.
+For example, if you are using OpenSSL, run the following command:
+
+```command
+openssl s_client -connect <address_id>.tmprl.cloud:7233 -showcerts -cert ~/certs/path.pem -key .~/certs/path.key -tls1_2
+```
+
+**Self-signed certificate**
+
+If you are using a self-signed certificate, run the following `tctl` command:
+
+```command
+tctl --namespace <namespace_id> --address <address_id>.tmprl.cloud:7233 --tls-cert-path ~/certs/path.pem --tls-key-path ~/certs/path.key namespace describe
+```
+
+### Renew TLS certification
+
+If the certificate has expired or is about to expire, the next step is to renew it.
+
```

Comment:
with tcld customer can list their certificates and the expiration date, not sure if it helps 
e.g `tcld namespace accepted-client-ca list --namespace antonio.a2dd6`